src/gui/vector/qgsattributesformmodel.cpp

```diff
@@ -1666,3 +1666,69 @@ void QgsAttributesFormLayoutModel::insertChild( const QModelIndex &parent, int r
   itemForIndex( parent )->insertChild( row, std::move( item ) );
   endInsertRows();
 }
+
+
+QgsAttributesFormProxyModel::QgsAttributesFormProxyModel( QObject *parent )
+  : QSortFilterProxyModel( parent )
+{
+}
+
+void QgsAttributesFormProxyModel::setAttributesFormSourceModel( QgsAttributesFormModel *model )
+{
+  mModel = model;
+  QSortFilterProxyModel::setSourceModel( mModel );
+}
+
+const QString QgsAttributesFormProxyModel::filterText() const
+{
+  return mFilterText;
+}
+
+void QgsAttributesFormProxyModel::setFilterText( const QString &filterText )
+{
+  // Since we want to allow refreshing the filter when, e.g.,
+  // users swtich to aliases, then we allow this method to be
+  // executed even if previous and new filters are equal
```

Comment:
This looks wrong -- if your model is well behaving, it should be emitting data changed signals when the display text is changed from name to alias, and the proxy model will automatically respond accordingly.

Reply:
It is in fact emitting `dataChanged()`, see https://github.com/qgis/QGIS/blob/17734b3a4661e73af3408b0663636a63fc3f96cb/src/gui/vector/qgsattributesformmodel.cpp#L457-L461

Not sure what I could be missing then. The proxy model doesn't seem to reflect the change unless we call `QgsAttributesFormProperties::updateFilteredItems()`.

Reply:
Hmm - i would try enabling the model test on it then, and see if any issues are reported. See

https://github.com/qgis/QGIS/blob/master/src/gui/layertree/qgslayertreeview.cpp#L37-L39
https://github.com/qgis/QGIS/blob/master/src/gui/layertree/qgslayertreeview.cpp#L107-L109
https://github.com/qgis/QGIS/blob/master/CMakeLists.txt#L761

Reply:
Thanks, done in https://github.com/qgis/QGIS/pull/61830/commits/d7104b704cac47c504f953d660a320ca3f1712a8
I haven't seen any issues from there.

Reply:
What if you call `setDynamicSortFilter( true );` in the proxy constructor?

Reply:
Still the same behavior, nothing seems to change calling that method.

Reply:
@nyalldawson, how do you think we could proceed here?

Reply:
Can you re-add setDynamicSortFilter( true ); ?

Reply:
> Can you re-add setDynamicSortFilter( true ); ?

Forgot to say that I attempted adding that line but nothing changes. In fact, `true` seems to be the default value for that property anyways.

More findings. With some filters (e.g., `n` or `g`), the update is triggered as expected. But with others like e.g., `k` it's not, weird...

https://github.com/user-attachments/assets/6a6055d6-3ae5-480d-8d53-f62e2441fb8b

Reply:
@nyalldawson, thanks for the suggestions!

Doing some tests, I figured out that any of these 3 solutions works:

  1) Call `dataChanged` recursively.

  2) Instead of emitting `dataChanged`, call these 2 lines from `QgsAttributesFormModel::setShowAliases()`:

         emit layoutAboutToBeChanged();
         emit layoutChanged();

  3) When the user toggles `showAliases`, invalidate the proxy models from `QgsAttributesFormProperties::toggleShowAliases()`:

         mAvailableWidgetsProxyModel->invalidate();
         mFormLayoutProxyModel->invalidate();

Do we stick to solution 1) ?